include_gif/src/lib.rs

```diff
@@ -1,25 +1,103 @@
 extern crate proc_macro;
 
 use proc_macro::TokenStream;
+use std::collections::HashSet;
 use std::fs::File;
 use std::io::Write;
+use syn::{parse_macro_input, Ident, LitStr};
+
+enum BppFormat {
+    Bpp1 = 0,
+    Bpp2 = 1,
+    Bpp4 = 2,
+}
+
+enum GlyphType {
+    Bagl,
+    Nbgl,
+}
+
+enum Input {
+    FileNameOnly(LitStr),
+    FileNameAndType(LitStr, GlyphType),
+}
+
+impl syn::parse::Parse for Input {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        // Parse the filename
+        let filename = input.parse::<LitStr>()?;
+        if input.is_empty() {
+            Ok(Input::FileNameOnly(filename))
+        } else {
+            // Parse comma separator (won't be used, just to skip it)
+            let _: syn::Token![,] = input.parse()?;
+            // Parse the glyph type
+            let glyph_type = input.parse::<Ident>()?;
+            match glyph_type.to_string().as_str() {
+                "BAGL" => Ok(Input::FileNameAndType(filename, GlyphType::Bagl)),
+                "NBGL" => Ok(Input::FileNameAndType(filename, GlyphType::Nbgl)),
+                _ => Err(syn::Error::new_spanned(glyph_type, "Invalid glyph type")),
+            }
+        }
+    }
+}
 
 #[proc_macro]
 pub fn include_gif(input: TokenStream) -> TokenStream {
-    let filename = syn::parse_macro_input!(input as syn::LitStr);
+    let input = parse_macro_input!(input as Input);
+    match input {
+        Input::FileNameOnly(filename) => {
+            // Default to Bagl if no type is specified.
+            let generate_type = GlyphType::Bagl;
+            generate_glyph(filename, generate_type)
+        }
+        Input::FileNameAndType(filename, generate_type) => generate_glyph(filename, generate_type),
+    }
+}
+
+fn generate_glyph(filename: LitStr, glyph_type: GlyphType) -> TokenStream {
+    let filename = filename.value();
     let mut decoder = gif::DecodeOptions::new();
     decoder.set_color_output(gif::ColorOutput::Indexed);
 
     let path = format!(
         "{}/{}",
         std::env::var("CARGO_MANIFEST_DIR").unwrap(),
-        filename.value()
+        filename
     );
     let file = File::open(path).unwrap();
     let mut decoder = decoder.read_info(file).unwrap();
 
     let frame = decoder.read_next_frame().unwrap().unwrap().clone();
     let palette = decoder.palette().unwrap();
+    let mut vec_output = Vec::new();
+
+    match glyph_type {
+        GlyphType::Bagl => {
+            let packed = generate_bagl_glyph(&frame, &palette);
+            write!(
+                &mut vec_output,
+                "(&{:?}, {}, {})",
+                packed, frame.width, frame.height
+            )
+            .unwrap();
+        }
+        GlyphType::Nbgl => {
+            let (compressed_buffer, bpp) = generate_nbgl_glyph(&frame, &palette);
```

Comment:
Does it mean that we only support compressed glyphs ? Maybe I am wrong, but it seems to me that some glyphs shall be supported in uncompressed format...

Reply:
I'm not sure there are benefits to uncompressed glyphs... they just take more memory :man_shrugging: . We could have options, I'm not against it, but not sure it would be really useful :thinking:

Reply:
Yes I agree with you regarding the "benefit" of using uncompressed glyphs but I remark this in the C SDK python script: https://github.com/LedgerHQ/ledger-secure-sdk/blob/e55673807f200645d2aacbd87dbfbb514b6df56b/lib_nbgl/tools/icon2glyph.py#L351C16-L351C72, so it may be relevant to consider it maybe 🤔

Reply:
@yogh333 indeed. Do you prefer we keep the same way of doing in things as in the C SDK script (using a `_nocomp` suffix on file names) or we add a new value for the 2nd parameter of our macro such as 'NBGL_NOCOMP' ? I would maybe prefer the second option. :thinking:

Reply:
ok for the second option